dbschema/engagement.esdl

```diff
@@ -130,8 +130,8 @@ module default {
       insert Engagement::CertificationCeremony {
         createdAt := datetime_of_statement(),
         modifiedAt := datetime_of_statement(),
-        createdBy := assert_exists(currentUser),
-        modifiedBy := assert_exists(currentUser),
+        createdBy := assert_exists(global currentActor),
```

Comment:
The current user is the current actor, if it is a user, otherwise null. current actor can be a system agent. created/modifiedBy is an actor type so we use that here to be the most accepting. current user should be used sparingly probably, when a user type is required.